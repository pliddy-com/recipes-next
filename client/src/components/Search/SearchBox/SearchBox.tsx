import {
  useCurrentRefinements,
  useSearchBox,
  UseSearchBoxProps
} from 'react-instantsearch-hooks-web';

import InputAdornment from '@mui/material/InputAdornment';
import Paper from '@mui/material/Paper';
import TextField from '@mui/material/TextField';

import SearchIcon from '@mui/icons-material/Search';

/* istanbul ignore next */
const queryHook: UseSearchBoxProps['queryHook'] = (query, search) => {
  search(query);
};

const SearchBox = () => {
  const { refine } = useSearchBox({
    queryHook
  });

  const refinements = useCurrentRefinements();

  return (
    <Paper
      className="search"
      component="form"
      elevation={0}
      onSubmit={(e) => {
        e.preventDefault();
      }}
      role="search"
      square
    >
      <TextField
        autoComplete="false"
        autoFocus={true}
        fullWidth
        inputProps={{ 'aria-label': 'search' }}
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <SearchIcon color="primary" />
            </InputAdornment>
          )
        }}
        id="search"
        name="search"
        onChange={(e) => refine(e.currentTarget.value)}
        placeholder="Search recipesâ€¦"
        size="small"
        value={refinements.items[0]}
        variant="outlined"
      />
    </Paper>
  );
};

export default SearchBox;
