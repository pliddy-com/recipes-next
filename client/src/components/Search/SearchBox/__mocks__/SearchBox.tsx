jest.createMockFromModule('../SearchBox');

export const SearchBox = jest.fn(() => (
  <div data-testid="search-box">SearchBox</div>
));

export default SearchBox;
