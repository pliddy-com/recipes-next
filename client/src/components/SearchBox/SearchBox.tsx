/* istanbul ignore file */
import IconButton from '@mui/material/IconButton';
import InputBase from '@mui/material/InputBase';
import SearchIcon from '@mui/icons-material/Search';
import Paper from '@mui/material/Paper';

import { connectSearchBox } from 'react-instantsearch-dom';
import { SearchBoxProvided } from 'react-instantsearch-core';

const MuiSearchBox = ({ currentRefinement, refine }: SearchBoxProvided) => (
  <Paper
    action=""
    component="form"
    role="search"
    square
    variant="outlined"
    sx={{
      p: '2px 4px',
      mb: '2rem',
      display: 'flex',
      alignItems: 'center',
      maxWidth: '400px'
    }}
  >
    <InputBase
      sx={{ ml: 1, flex: 1 }}
      placeholder="Searchâ€¦"
      inputProps={{ 'aria-label': 'search recipes' }}
      value={currentRefinement}
      onChange={(e) => refine(e.currentTarget.value)}
      autoComplete="false"
    />
    <IconButton type="button" sx={{ p: '10px' }} aria-label="search">
      <SearchIcon />
    </IconButton>
  </Paper>
);

const CustomSearchBox = connectSearchBox(MuiSearchBox);

const SearchBox = () => <CustomSearchBox />;

export default SearchBox;
