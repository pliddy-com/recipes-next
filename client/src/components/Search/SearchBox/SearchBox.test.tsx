// import testing-library methods
import { fireEvent, render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import SearchBox from './SearchBox';

jest.mock('react-instantsearch-hooks-web', () => ({
  useSearchBox: jest.fn().mockReturnValue({
    refine: jest.fn()
  }),
  useCurrentRefinements: jest.fn().mockReturnValue({
    items: [{ value: 1 }]
  })
}));

describe('SearchBox', () => {
  describe('when there is properly formatted content', () => {
    it('it renders the SearchBox', () => {
      const { asFragment, queryByRole } = render(<SearchBox />);

      const form = queryByRole('search');
      const textField = queryByRole('textbox') as HTMLInputElement;

      expect(form).toBeInTheDocument();
      expect(textField).toBeInTheDocument();

      form && fireEvent.submit(form);

      textField && fireEvent.change(textField, { target: { value: 'test' } });

      // assert that the component matches the existing snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
