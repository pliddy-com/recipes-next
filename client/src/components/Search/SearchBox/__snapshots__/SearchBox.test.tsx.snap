// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox when there is properly formatted content it renders the SearchBox 1`] = `
<DocumentFragment>
  <form
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 search css-9mq7az-MuiPaper-root"
    role="search"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <input
          aria-invalid="false"
          aria-label="search"
          autocomplete="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
          id="search"
          name="search"
          placeholder="Search recipes…"
          type="text"
          value="[object Object]"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
